src/atomate2/ase/schemas.py

```diff
@@ -0,0 +1,479 @@
+"""Schema definitions for Atomic Simulation Environment (ASE) tasks.
+
+The following code has been taken and generalized to
+generic ASE calculators from
+https://github.com/materialsvirtuallab/m3gnet
+The code has been released under BSD 3-Clause License
+and the following copyright applies:
+Copyright (c) 2022, Materials Virtual Lab.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Optional, Union
+
+from ase.stress import voigt_6_to_full_3x3_stress
+from ase.units import GPa
+from emmet.core.math import Matrix3D, Vector3D
+from emmet.core.structure import MoleculeMetadata, StructureMetadata
+from emmet.core.utils import ValueEnum
+from emmet.core.vasp.calculation import StoreTrajectoryOption
+from pydantic import BaseModel, Field
+from pymatgen.core import Molecule, Structure
+from pymatgen.core.trajectory import Trajectory as PmgTrajectory
+
+_task_doc_translation_keys = {
+    "input",
+    "output",
+    "ase_calculator_name",
+    "dir_name",
+    "included_objects",
+    "objects",
+    "is_force_converged",
+    "energy_downhill",
+}
+
+
+class AseResult(dict):
+    """Schema to store outputs in ForceFieldTaskDocument."""
+
+    def __init__(self, **kwargs: Any) -> None:
+        kwargs = {
+            "final_ionic_config": None,
+            "trajectory": None,
+            "is_force_converged": None,
+            "energy_downhill": None,
+            "dir_name": None,
+            **kwargs,
+        }
+        super().__init__(**kwargs)
+
+
+class AseObject(ValueEnum):
+    """Types of ASE data objects."""
+
+    TRAJECTORY = "trajectory"
+
+
+class AseBaseModel(BaseModel):
+    """Base document class for ASE input and output."""
+
+    ionic_configuration: Optional[Union[Structure, Molecule]] = Field(
```

Comment:
I think `mol_or_struct` is more informative / good implicit type hinting. I don't think we can change IonicStep to avoid breaking changes on the forcefield side of things